### ISSUE TITLE
The version of the data set and its layout

### ISSUE THREAD

**Issue by hhy12336** [edornd/agrivision-2022#2]
Hi, I really liked your project, but I had some issues with the version and distribution of the dataset when I reproduced it.
I did not find the agriculture-vision2020 dataset, I used the agriculture-vision2021 dataset. 
I found that your database file uses the 'gt' folder and I don't know what data this is.
`data = dict(samples_per_gpu=4,
            workers_per_gpu=4,
            train=dict(type=dataset_type,
                       data_root=data_root,
                       img_dir='train/images/rgb',
                       nir_dir='train/images/nir',
                       ann_dir='train/gt',
                       pipeline=train_pipeline),
            val=dict(type=dataset_type,
                     data_root=data_root,
                     img_dir='valid/images/rgb',
                     nir_dir='valid/images/nir',
                     ann_dir='valid/gt',
                     pipeline=test_pipeline),
            test=dict(type=dataset_type,
                      data_root=data_root,
                      img_dir='valid/images/rgb',
                      nir_dir='train/images/nir',
                      ann_dir='valid/gt',
                      pipeline=test_pipeline))`

The layout of my dataset folder is as follows:
Agriculture-Vision-2021

> test

       -boundaries
       -images
       -masks
> train

       -boundaries
       -images
       -labels  (Contains labels of 9 categories)
       -masks
> val

      -boundaries
      -images
      -labels  (Contains labels of 9 categories)
      -masks

I would like to know what version of the data set you are using, or how to set the path of ‘gt’. Thank you very much for taking time out of your busy schedule to answer my questions. Thank you very much.

**Comment by edornd** [comment]
Hi, yes the AV-2021 version should be fine! Regarding the structure, first have a look at the `mmseg` documentation [here](https://mmsegmentation.readthedocs.io/en/latest/advanced_guides/add_datasets.html), this should give you an idea on why it's formatted that way.
The `gt` folder stands for _ground truth_, and it's generated by merging each binary label (+ the boundary map) in a single image, apologies for the lack of documentation.
In short, you will need to gather all the masks in each subfolder, match them by tile ID, then create a new image where each pixel has a single index. This index is either the index of the specific category it belongs to, or 255 if outside the boundaries.
I realize now that the script I originally used is not present, I'll upload it and link the commit to this issue!

**Comment by edornd** [comment]
The `process` function is the minimum requirement, if I recall correctly, the rest can be customized. Unfortunately I don't remember much more than that, I hope it's enough!

**Comment by Akivilio** [comment]
Thank you very much for your help. Fortunately, the project is worked.

**Comment by edornd** [comment]
Closing this, feel free to open a new one if you need!
